minicluster/src/main/java/tachyon/master/LocalTachyonClusterMultiMaster.java

```diff
@@ -185,35 +160,50 @@ private void mkdir(String path) throws IOException {
     }
   }
 
-  public void start() throws IOException {
+  @Override
+  protected void startWorker(TachyonConf conf) throws IOException {
+    mWorkerConf = WorkerContext.getConf();
+    mWorkerConf.merge(conf);
+
+    // Setup conf for worker
     int numLevels = 1;
-    mTachyonHome =
-        File.createTempFile("Tachyon", "U" + System.currentTimeMillis()).getAbsolutePath();
-    mWorkerDataFolder = "/datastore";
+    mWorkerConf.set(Constants.WORKER_TIERED_STORE_LEVELS, Integer.toString(numLevels));
+
+    for (int level = 1; level < numLevels; level ++) {
+      String tierLevelDirPath =
+          String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level);
+      String[] dirPaths = mWorkerConf.get(tierLevelDirPath).split(",");
+      String newPath = "";
+      for (String dirPath : dirPaths) {
+        newPath += mTachyonHome + dirPath + ",";
+      }
+      mWorkerConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level),
+          newPath.substring(0, newPath.length() - 1));
+    }
+
+    mWorkerConf.set(Constants.WORKER_BIND_HOST, mHostname);
+    mWorkerConf.set(Constants.WORKER_DATA_BIND_HOST, mHostname);
+    mWorkerConf.set(Constants.WORKER_WEB_BIND_HOST, mHostname);
+    mWorkerConf.set(Constants.WORKER_WORKER_BLOCK_THREADS_MIN, "1");
+    mWorkerConf.set(Constants.WORKER_WORKER_BLOCK_THREADS_MAX, "100");
+
+    runWorker();
+    // The client context should reflect the updates to the conf.
+    ClientContext.reset(mWorkerConf);
+  }
 
-    mHostname = NetworkAddressUtils.getLocalHostName(100);
+  @Override
+  protected void setupTest(TachyonConf conf) throws IOException {}
 
-    mMasterConf = MasterContext.getConf();
-    mMasterConf.set(Constants.IN_TEST_MODE, "true");
-    mMasterConf.set(Constants.TACHYON_HOME, mTachyonHome);
+  @Override
+  protected void startMaster(TachyonConf conf) throws IOException {
+    mMasterConf = conf;
     mMasterConf.set(Constants.ZOOKEEPER_ENABLED, "true");
```

Comment:
extract extra setting into overrided newTestConf but the zookeeper related config still in the LocalTachyonClusterMultiMaster.java